functions/src/models/offers/index.ts

```diff
@@ -1,8 +1,8 @@
 import { FirestoreDataConverter } from '@google-cloud/firestore';
 import { IsEnum, IsNotEmpty, IsObject, IsString, ValidateNested } from 'class-validator';
-import { firestore } from 'firebase-admin';
+import { firestore } from 'firebase';
```

Comment:
This was my first approach at solving the lint errors.
But because we are using the `firestore` instance from `firebase-admin`, I am not sure if this would cause issues in the cloud function? maybe we would have to use `Type Generics` here and have a function that would use appropriate type based on if it's a test and whether the test is demanding client or server. I think that would be a better solution. but for now wouldn't the existing solution be enough